app/src/main/java/io/homeassistant/companion/android/settings/SettingsPresenterImpl.kt

```diff
@@ -39,6 +40,7 @@ class SettingsPresenterImpl @Inject constructor(
             when (key) {
                 "location_zone" -> integrationUseCase.setZoneTrackingEnabled(value)
                 "location_background" -> integrationUseCase.setBackgroundTrackingEnabled(value)
+                "fullscreen" -> integrationUseCase.setFullScreenEnabled(value)
                 else -> throw Exception()
             }
             settingsView.onLocationSettingChanged()
```

Comment:
This is being called when you toggle full screen and it is causing a prompt for location settings.  The `settingsView.onLocationSettingChanged()` should only be called when a location settings has changed, not whenever a boolean setting changed.

Reply:
Add condition for settingsView.onLocationSettingChanged() call